### PR TITLE
Update DG with bill command, date range, model diagram, and acknowledgements

### PR DESCRIPTION
**Pull request by JustinPeng13**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -231,7 +231,38 @@ The following activity diagrams summarizes what happens when a user enters an `e
...
+### Bill Command
+
+#### Implementation:
+* The `bill` command takes in an INDEX indicating the index of the guest to edit in the current panel (starting from 1) and the `bill` field and is supported by the `BillCommandParser` that extracts the bill value.
+
+The following activity diagrams summarizes what happens when a user enters a `bill` command.
+
+![BillActivityDiagram](images/BillActivityDiagram.png)
+
+#### Design Considerations:
```
Comment (bryanhce) [inline docs/DeveloperGuide.md:281]: Do you think its necessary to mention the "b/" prefix in the bill command?

**Review by bryanhce — COMMENTED** [review]
LGTM, just 1 question!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (JustinPeng13) [reply docs/DeveloperGuide.md:281]: Yes, added

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -228,10 +228,44 @@ The following activity diagrams summarizes what happens when a user enters an `e
...
+and that minimal calculation is needed to reset the bill to 0 (`b/-CURRENT_VALUE`), we decided to proceed with Alternative 2.
+
+**Aspect: Accepting negative values**
+* As the user may want to give the guest a discount or fix an erroneous charge, we allow the user to input a negative value.
+
+**Aspect: Accepting signed and unsigned positive values**
+* As the user may prefer to see `+` differentiating positive and negative values, or leave out the `+` sign for convenience, we decided to accept both formats for positive values.
+
+**Aspect: Using the `b/` prefix**
+* To standardise the formatting and testing for field inputs, we decided to include the `b/` prefix in the command syntax
```
Comment (bryanhce) [inline docs/DeveloperGuide.md]: Should this sentence end with a full stop?

**Review by bryanhce — COMMENTED** [review]
LGTM

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (JustinPeng13) [reply docs/DeveloperGuide.md]: Yes, added

**Review by bryanhce — APPROVED** [review]
LGTM

**Pull request merged**